src/ansys/additive/core/simulation_task.py

```diff
@@ -0,0 +1,315 @@
+# Copyright (C) 2023 - 2024 ANSYS, Inc. and/or its affiliates.
+# SPDX-License-Identifier: MIT
+#
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+"""Provides progress updates."""
+
+from __future__ import annotations
+
+import os
+import zipfile
+
+from ansys.api.additive.v0.additive_domain_pb2 import ProgressState as ProgressMsgState
+from ansys.api.additive.v0.additive_operations_pb2 import OperationMetadata
+from ansys.api.additive.v0.additive_simulation_pb2 import SimulationResponse
+from google.longrunning.operations_pb2 import (
+    GetOperationRequest,
+    ListOperationsRequest,
+    Operation,
+    WaitOperationRequest,
+)
+from google.protobuf.any_pb2 import Any
+from google.protobuf.duration_pb2 import Duration
+
+from ansys.additive.core.download import download_file
+from ansys.additive.core.microstructure import MicrostructureInput, MicrostructureSummary
+from ansys.additive.core.microstructure_3d import Microstructure3DInput, Microstructure3DSummary
+from ansys.additive.core.porosity import PorosityInput, PorositySummary
+from ansys.additive.core.progress_handler import (
+    DefaultSingleSimulationProgressHandler,
+    IProgressHandler,
+    Progress,
+    ProgressState,
+)
+from ansys.additive.core.server_connection import ServerConnection
+from ansys.additive.core.simulation import SimulationError
+from ansys.additive.core.single_bead import SingleBeadInput, SingleBeadSummary
+from ansys.additive.core.thermal_history import ThermalHistoryInput, ThermalHistorySummary
+
+
+class SimulationTask:
+    def __init__(
+        self,
+        server_connections: list[ServerConnection],
```

Comment:
With redesign of simulation_task only having one long-running operation and one server, along with moving nsims_per_server enforcment to the server side, this will be much easier to maintain.